frontend/components/TableContainer/DataTable/DataTable.tsx

```diff
@@ -46,14 +46,15 @@ interface IDataTableProps {
   primarySelectActionButtonText?: string | ((targetIds: number[]) => string);
   onPrimarySelectActionClick: any; // figure out type
   secondarySelectActions?: IActionButtonProps[];
-  onSelectSingleRow?: (value: Row) => void;
-  onResultsCountChange?: (value: number) => void;
   isClientSidePagination?: boolean;
   isClientSideFilter?: boolean;
   highlightOnHover?: boolean;
   searchQuery?: string;
   searchQueryColumn?: string;
   selectedDropdownFilter?: string;
+  clearSelectionCount?: number;
```

Comment:
I'm not crazy about adding another prop to this component, but from what I could tell none of the existing props give us the ability to clear all selections from outside the data table. Right now only the "Clear" button does that and it's inside the component.

This is a good example for the need to refactor DataTable. I spent a good amount of time trying to work with toggleAllPagesSelected as it seems like that should do what we want but it doesn't appear to have access to the function I need.